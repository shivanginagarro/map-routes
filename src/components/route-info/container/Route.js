import React from 'react';
import '../route-style.css';
import RouteForm from './RouteForm';


class Route extends React.PureComponent {
//renders routefrom 
    render() {
        return (
            <div id="my-route" className={"route-info-wrapper"}>
                <RouteForm getRoutes={this.props.getRoutes} {...this.props} />
            </div>
        )
    }
}

export default Route;