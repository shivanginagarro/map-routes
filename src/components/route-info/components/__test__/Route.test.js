import React from "react";
import Route from "../Route";
import renderer from "react-test-renderer"

describe("Component:Route", () => {
    it("Testing the Parent Route Component", () => {
        let informationDetail = {
            error: '',
            total_distance: '',
            total_time: ''
        }
        let getRoutes = () => {
            console.log('inside routes testing');
        }
        let reset = () => {
            console.log("inside reset testing");
        }
        const tree = renderer.create(<Route informationDetail={informationDetail} getRoutes={getRoutes} reset={reset} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
})