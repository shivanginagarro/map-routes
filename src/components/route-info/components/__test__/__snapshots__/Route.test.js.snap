// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:Route Testing the Parent Route Component 1`] = `
<div
  className="route-info-wrapper"
  id="my-route"
>
  <div
    className="directions-form"
  >
    <label
      className="form-label"
    >
       
      <span>
        Starting Point :
      </span>
        
      <input
        name="starting"
        placeholder="Pick Up From"
        type="text"
      />
       
    </label>
    <label
      className="form-label"
    >
       
      <span>
        Dropping Point :
      </span>
        
      <input
        name="dropping"
        placeholder="Drop To"
        type="text"
      />
       
    </label>
    <div
      className="btn-container"
    >
      <button
        className="form-btn"
        onClick={[Function]}
      >
        Submit
      </button>
       
      <button
        onClick={[Function]}
      >
        Reset
      </button>
    </div>
  </div>
</div>
`;
